### ISSUE TITLE
use API functions for setting Apache port

### ISSUE THREAD

**Issue by escherlat** [engintron/engintron#134]
engintron.sh calls the whostmgr2 binary with the --updatetweaksettings as part of setting the Apache port. That can cause undesirable behavior on systems. 'whostmgr2 --updatetweaksettings' is designed to only be run on a freshly installed cPanel & WHM system.

Rather than calling the whostmgr2 binary, and rather than manually modifying cpanel.config, use the set_tweaksetting API function with the whmapi1 binary:

`$ /usr/local/cpanel/bin/whmapi1 set_tweaksetting key=apache_port value=0.0.0.0:8080                                                                                                
--- 
metadata: 
  command: set_tweaksetting
  reason: OK
  result: 1
  version: 1
`

**Comment by fevangelou** [comment]
2nd cPanel folk here :) Welcome Kenneth and thanks for the tip. I'll update the codebase.

Should I assume that reading the config file in order to determine the existing port in Apache be changed to use the API as well?

**Comment by escherlat** [comment]
When you say "reading the config file" are you referring to httpd.conf or cpanel.config? If cpanel.config, then yes please use our API (get_tweaksetting).

Please use our APIs as much as possible. That leaves us (cPanel) free to change implementations without breaking 3rd party applications :)

If you find we don't provide an API function for something, please don't hesitate to reach out to us. Usually adding API functions is simple and we enjoy helping the development community around us.

**Comment by fevangelou** [comment]
Yes, I was referring to cpanel.config indeed.

As I mentioned in another ticket, it would be great if you could expose the current configuration as well as other basic system info directly as json data or a PHP object, accessible via the root (or any "all" privileges) user.

The current implementations using the token won't really work when a plugin using PHP tries to access say JSON or XML data coming from the API. That's because PHP runs from the server and therefore the authentication token used from the user cannot be used by PHP.

In other words, I can't just do json_decode(file_get_contents("URL_to_JSON_API")) from a PHP file in a WHM plugin. This fails to authenticate as the URL is retrieved from the server, not the user's browser.

In the case of checking for "all: 1" in privileges, I had to "grep" for that value running shell_exec and retrieving the data using a shell command.

Unless there's a way to create a token for PHP in cases like the above and I just didn't see it on your docs site.
